src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -136,21 +137,22 @@ def import_config(cmd,
         src_kvs = __read_kv_from_app_service(
             cmd, appservice_account=appservice_account, prefix_to_add=prefix, content_type=content_type)
 
-    # if customer needs preview & confirmation
-    if not yes:
+    if strict or not yes:
         # fetch key values from user's configstore
         dest_kvs = __read_kv_from_config_store(azconfig_client,
```

Comment:
>dest_kvs [](http://example.com/codeflow?start=8&length=8)

For strict mode, we should query `dest_kvs` with the supplied `prefix`, not `SearchFilterOptions.ANY_KEY`. For preview, its okay to fetch all keys, but for deleting, we only want to delete those keys that match the given prefix.

Reply:
I had totally missed this case, now fixed.